src/execution/as_abi.rs

```diff
@@ -613,3 +686,90 @@ fn alloc_string_array(env: &ASEnv, vec: &[String]) -> ABIResult<i32> {
         Err(err) => abi_bail!(err),
     }
 }
+
+#[allow(dead_code)]
+/// Tooling, return a buffer (Vec<u8>) from a given offset
+fn get_buffer(memory: &Memory, ptr: i32) -> ABIResult<Vec<u8>> {
+    match BufferPtr::new(ptr as u32).read(memory) {
+        Ok(buffer) => Ok(buffer),
+        Err(err) => abi_bail!(err),
+    }
+}
+
+/// Flatten a Vec<Vec<u8>> to a Vec<u8> with the format:
+/// L (32 bits LE) V1_L (8 bits) V1 (8bits * V1_L), V2_L ... VN (8 bits * VN_L)
+fn ser_bytearray_vec(data: &Vec<Vec<u8>>, max_entry: usize) -> ABIResult<Vec<u8>> {
+    if data.is_empty() {
+        return Ok(Vec::new());
```

Comment:
is that what we want ? Don't we want to encore "length 0" to avoid generating a special case ?

Reply:
It avoids to write 4 u8 (u32) set to 0. And I think we should test this edge case in massa-as-sdk, in order to avoid to blindly trust massa-sc-runtime (against future accidental changes)?

Reply:
It's a balance between the added ser compacity in the "0-element" edge case and the added code complexity of handling that special case apart. My personal preference in this kind of cases goes to always sending the length byte and simplifying code (no edge cases to handle on all sides). But it's purely a personal preference.